geniza/corpus/management/commands/googledoc_to_annotation.py

```
@@ -0,0 +1,389 @@
+import unicodedata
+from collections import defaultdict
+
+from addict import Dict
+from bs4 import BeautifulSoup
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.template.defaultfilters import pluralize
+from django.utils import timezone
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaIoBaseDownload
+from parasolr.django.signals import IndexableSignalHandler
+from rich.progress import MofNCompleteColumn, Progress
+
+from geniza.annotations.models import Annotation
+from geniza.corpus.annotation_export import AnnotationExporter
+from geniza.corpus.management.commands import tei_to_annotation
+from geniza.corpus.models import Document
+from geniza.footnotes.models import Footnote, Source
+
+
+class Command(tei_to_annotation.Command):
+    """Synchronize Google Doc translations to digital translation footnote content"""
+
+    v_normal = 1  # default verbosity
+
+    normalized_unicode = set()
+
+    document_not_found = []
+
+    source_not_found = []
+
+    REMOVE_ATTRIBUTES = ["style", "class"]  # attributes to strip from html elements
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "folder_id",
+            help="The ID of the Google Drive folder containing translation documents.",
+        )
+
+    def handle(self, *args, **options):
+        self.verbosity = options["verbosity"]
+        self.stats = defaultdict(int)
+        self.stats["files"] = 0
+        self.stats["created"] = 0
+        self.script_run_start = timezone.now()
+
+        # get script user for log entries
+        self.script_user = User.objects.get(username=settings.SCRIPT_USERNAME)
+
+        # get drive id from settings
+        self.drive_id = settings.GOOGLE_DRIVE_ID
```

should raise a command error if not configured with instructions to add